resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -104,6 +104,13 @@ private[spark] class KubernetesClusterSchedulerBackend(
     conf.getOption("spark.app.id").getOrElse(appId)
   }
 
+  override def getDriverAttributes: Option[Map[String, String]] = Some(Map(
```

Comment:
This part looks a little suspicious (although I didn't verify deeply).

Reply:
I have replaced this with `"SPARK_DRIVER_ATTRIBUTE_*" environment variables, similar to `"SPARK_EXECUTOR_ATTRIBUTE_". When `spark.ui.custom.driver.log.url` is defined, some driver attribute env vars are added just like in 
#47692.